cmd/protoc-gen-go-grpc/gengogrpc/grpc.go

```diff
@@ -0,0 +1,415 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package gengogrpc contains the gRPC code generator.
+//
+// This package is exported to permit github.com/golang/protobuf/protoc-gen-go
+// to depend on it. No other packages should depend on it.
```

Comment:
So the plan is to delete the code there and reference this code instead?

This has pros and cons:
- Pro: v1 and v2 codegen do not need to be maintained separately
- Con: changes here will impact the v1 codegen, meaning we need to test the v1 codegen (in the other repo) if we want to make changes here.

Might it be better to just have separate implementations and allow them to drift?  (Also: clearly marking the v1 codegen as deprecated and no longer receiving new features.)

Reply:
My plan had been to delete the generator from `github.com/golang/protobuf` and reference the one here. If you'd rather we deprecate that one and let it drift, I'm fine with that approach. What's your preference?

Reply:
Not sure that the con is as significant, since `github.com/golang/protobuf` can specify that they depend on a specific version of the generator herein. Since v1 is not meant to be receiving new features, they should not advance to newer versions of the grpc gen, except for important bug fixes that advance only the minor version number. This should reduce the need for cross-repo testing to those bug fixes that are deemed necessary to backport to v1. That said, I think both strategies are fine.